### PR TITLE
add config to limit ShellBolt internal _pendingWrites queue length

### PR DESCRIPTION
**Pull request by xiaokang**
to fix #541, replace #555

### CODE REVIEW

File: storm-core/src/jvm/backtype/storm/task/ShellBolt.java [hunk 1]
```diff
@@ -71,6 +72,10 @@ public ShellBolt(String... command) {
 
     public void prepare(Map stormConf, TopologyContext context,
                         final OutputCollector collector) {
+        Object maxPending = stormConf.get(Config.TOPOLOGY_SHELLBOLT_MAX_PENDING);
+        if (maxPending != null && maxPending instanceof Number) {
```
Comment (nathanmarz) [inline storm-core/src/jvm/backtype/storm/task/ShellBolt.java]: Don't need this instanceof check here. Other code will validate the config for schema errors.

**Comment by xiaokang** [comment]
Got it, Config.java already configured its schema. Removed instanceof check.

**Comment by nathanmarz** [comment]
+1 for 0.9.0

**Comment by ptgoetz** [comment]
+1

Added to 0.9.0 milestone

**Comment by xumingming** [comment]
+1

**Pull request merged**